model/src/main/kotlin/ScannerRun.kt

```diff
@@ -135,6 +135,20 @@ data class ScannerRun(
             }
         }
 
+        scanResults.getDuplicates { it.provenance to it.scanner }.keys.let { duplicates ->
+            require(duplicates.isEmpty()) {
+                val (dupProvenance, dupScanner) = duplicates.first()
+
+                buildString {
+                    appendLine("Found multiple scan results for the same provenance and scanner.")
+                    appendLine("Scanner:")
+                    appendLine(dupScanner.toYaml())
+                    appendLine("Provenance:")
+                    append(dupProvenance.toYaml())
+                }
```

Comment:
As these lines are all appended unconditionally, maybe hard-code as:

                """
                    Found multiple scan results for the same provenance and scanner.
                    Scanner:
                    ${dupScanner.toYaml()}
                    Provenance:
                    ${dupProvenance.toYaml()}
                """.trimIndent()

Reply:
Ok, then let's keep using `buildString`.